app/services/candidate_interface/sort_application_choices.rb

```diff
@@ -4,19 +4,28 @@ def self.call(application_choices:)
       scope = application_choices.from <<~WITH_GROUP.squish
         (
           SELECT application.*,
+          CASE
+            WHEN (status IN ('offer')) THEN 'offers_received'
+            WHEN (status IN ('unsubmitted', 'application_not_sent', 'cancelled')) THEN 'draft'
+            WHEN (status IN ('rejected', 'conditions_not_met')) THEN 'unsuccessful'
+            WHEN (status IN ('interviewing', 'inactive', 'awaiting_provider_decision')) THEN 'in_progress'
+            WHEN (status IN ('offer_withdrawn', 'withdrawn')) THEN 'withdraw'
+            WHEN (status IN ('declined')) THEN 'declined'
+            ELSE ''
+          END AS application_choices_group,
```

Comment:
I chosen consciously to add this to not rely on application_choices_group_number below. With application_choices_group_number made the code more difficult to read so I consciously add the application_choices_group (see commit history for more details). Putting the groups here made the code very easy to understand up to the `ApplicationChoiceListComponent` in my opinion.